pkg/detectors/aws/aws.go

```diff
@@ -115,12 +131,19 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				Raw:          []byte(resIDMatch),
 				Redacted:     resIDMatch,
 				RawV2:        []byte(resIDMatch + resSecretMatch),
+				ExtraData: map[string]string{
+					"resource_type": resourceTypes[resIDMatch[0:4]],
+				},
 			}
 
 			if verify {
 				verified, extraData, verificationErr := s.verifyMatch(ctx, resIDMatch, resSecretMatch, true)
 				s1.Verified = verified
-				s1.ExtraData = extraData
+				//Append the extraData to the existing ExtraData map.
+				// This will overwrite with the new verified values.
+				for k, v := range extraData {
```

Comment:
question: Did you consider passing `s1` or even just the `ExtraData` map to `verifyMatch` so we set the values when we are setting all the other ExtraData value within that function?

Reply:
We definitely could. I have another PR coming re: account number mapping. We can access AWS account numbers just based on the ID (AKIA*****) value. However, if we verify the secret, then we'll want to have the value from AWS override the locally computed one. (It should be correct the vast majority of the time, if not all the time, but just in case it's not, we can overwrite with the official number). For that reason, I though this would be simplest + involve changing fewer functions. If for style purposes, you'd like me to change it, I'm happy to.

Reply:
Nope, this is just fine. Thanks for the explanation.